### PR TITLE
#944 - Healing items for specific spells give no +healing effect

### PR DESCRIPTION
**Pull request by Leadballoon2000**
The item's bonus was being applied to each tick. Note that coefficient isn't correct for Rejuvantion - should be 0.2

https://github.com/cmangos/issues/issues/944

### CODE REVIEW

**Comment by killerwife** [comment]
Still need to fix those tabs ^^

File: src/game/Unit.cpp [hunk 1]
```diff
@@ -6229,18 +6230,17 @@ uint32 Unit::SpellHealingBonusDone(Unit* pVictim, SpellEntry const* spellProto,
             continue;
         switch ((*i)->GetModifier()->m_miscvalue)
         {
-            case 4415: // Increased Rejuvenation Healing
-            case 4953:
-            case 3736: // Hateful Totem of the Third Wind / Increased Lesser Healing Wave / LK Arena (4/5/6) Totem of the Third Wind / Savage Totem of the Third Wind
```
Comment (cala) [inline src/game/Unit.cpp]: Why remove that spell?

Comment (killerwife) [inline src/game/Unit.cpp]: He said its wotlk only.

Comment (robinsch) [inline src/game/Unit.cpp]: It's not. Check for SPELL_AURA_OVERRIDE_CLASS_SCRIPTS with miscvalue 3736

Comment (killerwife) [inline src/game/Unit.cpp]: Turns out you are right. Will tell him to change it back.

**Comment by killerwife** [comment]
@Leadballoon2000 remove the removal of the alleged wotlk thing, since its present in TBC too

**Comment by Leadballoon2000** [comment]
Sorry, I was wrong to say that it's a WotLK item. It went in during the last TBC season.
The item doesn't   have any +healing bonus so it doesn't need to have an entry in this case statement. I've tested that the +resilience effect still occurs with this change applied.

I meant to add a commit comment but I'm having trouble getting Git to do what I want so far.

**Comment by killerwife** [comment]
No worries, thats the whole point of reviewing stuff.

**Comment by Leadballoon2000** [comment]
I've added the case statement back since it's needed for item id 25645 Totem of the Plains
I don't see how the comment regarding the PvP relic relates to the statement.
There will need to be a database entry to spell_affect to get the Totem of the Plains to work, I'll raise an issue if there isn't one already.

**Comment by Leadballoon2000** [comment]
I've added a number of other items to this PR but haven't been able to handle them the same way since the items add a dummy aura which doesn't have a miscvalue set.

**Comment by Leadballoon2000** [comment]
I'm having trouble with Libram of Souls Redeemed itemid 28592
Blessing of Healing has two effect with the +healing values for Holy Light and Flash of Light. The aura for the libram only has one effect for Holy Light.
It doesn't seem very satisfactory to hard code the bonues based on the spell cast.

**Comment by robinsch** [comment]
https://github.com/robinsch/TBCPvP/commit/fb0427f31f2e25f24ef635f2d979043f8dc29ee4

**Comment by Leadballoon2000** [comment]
OK, I'll back out the Libram of Souls Redeemed change :)

Shouldn't the effect of the Libram and of Blessing of Light add to TakenAdvertisedBenefit otherwise downranked spells would get the full bonus?

**Comment by killerwife** [comment]
@Leadballoon2000 feel free to test out @robinsch patch, he isnt contributing to cmangos atm anyway, but I dont think he will mind a patch of his making it here.

**Comment by killerwife** [comment]
@Leadballoon2000 one more thing, do it properly so that robinshes authorship is kept for the second commit. Either do cherry-pick or commit --author

**Comment by Leadballoon2000** [comment]
OK, I'm (pretty sure) I've updated the author.

File: src/game/Unit.cpp [hunk 2]
```diff
@@ -1197,7 +1197,7 @@ void Unit::CastCustomSpell(Unit* Victim, SpellEntry const* spellInfo, int32 cons
 
     if (bp1)
         spell->m_currentBasePoints[EFFECT_INDEX_1] = *bp1;
-
+
```
Comment (Phatcat) [inline src/game/Unit.cpp]: this tab is a bit needless, wouldn't you say? =)

**Comment by cyberium** [comment]
@killerwife, lot of hardcoded stuff here. Is it still needed?

**Comment by killerwife** [comment]
@cyberium it is needed, but not in this form. Closing and making a github issue is the way to go propably.

**Pull request closed without merging**